code/contracts/ONEWallet.sol

```diff
@@ -38,16 +39,16 @@ contract ONEWallet {
         height = height_;
         interval = interval_;
         t0 = t0_;
-        lifespan_ = lifespan_;
+        lifespan_ = lifespan_; // IH: what are the expected values?
         lastResortAddress = lastResortAddress_;
         dailyLimit = dailyLimit_;
-        maxOperationsPerInterval = maxOperationsPerInterval_;
+        maxOperationsPerInterval = maxOperationsPerInterval_; // IH: what are the expected values?
     }
 
     function retire() external returns (bool)
     {
         require(uint32(block.timestamp) / interval - t0 > lifespan, "Too early to retire");
-        require(lastResortAddress != address(0), "Last resort address is not set");
+        require(lastResortAddress != address(0), "Last resort address is not set"); // IH: it needs method for adjustment; currently, if it adjust it in constructor to 0x0, retire() will never work
```

Comment:
Yes, I will add a function that allows setting  lastResortAddress iff it is 0

Reply:
fixed in https://github.com/polymorpher/one-wallet/commit/d289e3e3802c664ab73d97ec5aa79631f896af0f